topdown/http_test.go

```diff
@@ -2514,3 +2518,74 @@ var httpSendHelperRules = []string{
 		no_headers = object.remove(resp, ["headers"])
 	}`,
 }
+
+func httpSocket(path string) (net.Listener, error) {
+	sock, err := net.Listen("unix", path)
+	if err != nil {
+		return nil, err
+	}
+
+	// install signal handler to clean up
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
+	go func(c chan os.Signal) {
+		<-c
+		sock.Close()
+		os.Exit(0)
+	}(sigc)
+
+	return sock, nil
+}
+func TestHTTPWithSocket(t *testing.T) {
+	tmpF, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.FailNow()
+	}
+
+	socketPath := tmpF.Name()
+	tmpF.Close()
+	os.Remove(socketPath)
+
+	socket, err := httpSocket((socketPath))
+	if err != nil {
+		t.FailNow()
+	}
+
+	rs := http.Server{
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+		}),
+	}
+
+	go func() {
+		_ = rs.Serve(socket)
+	}()
+	defer rs.Close()
+
+	path := fmt.Sprintf("socket=%s", url.PathEscape(socketPath))
+	rawURL := fmt.Sprintf("unix://localhost/end/point?%s", path) // Send a request to the server over the socket
+	ok, rawURL, tr := useSocket(rawURL, nil)
+	if !ok {
+		t.Error("useSocket() returned !ok")
+		t.FailNow()
+	}
+
+	client := http.Client{
+		Transport: tr,
+	}
+
+	req, err := http.NewRequest("GET", rawURL, nil)
+	if err != nil {
+		t.FailNow()
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
```

Comment:
[nit] here and in other places, let's do
```suggestion
		t.Fatal(err)
```
☝️ has the same effect and is a little more consistent with this code base, I think (hope 🤞)